drivers/sdhc/rcar_mmc.c

```diff
@@ -827,14 +827,18 @@ static int rcar_mmc_sd_buf_rx_tx_data(const struct device *dev,
 		return -EINVAL;
 	}
 
+	start_data_xref_us = k_ticks_to_us_ceil32(k_uptime_ticks());
 	for (block = 0; block < data->blocks; block++) {
 		uint8_t *buf = (uint8_t *)data->data + (block * data->block_size);
 		uint32_t info2_reg;
 		uint16_t w_off; /* word offset in a block */
+		uint32_t elapsed_us = k_ticks_to_us_ceil32(k_uptime_ticks()) - start_data_xref_us;
+		int32_t left_time_us = data->timeout_ms * 1000 - elapsed_us;
```

Comment:
`elapsed_us` here will be converted into plain `int`, which will lead to interesting results in half of the cases :)

Reply:
nope, see C standard about integer promotion
`elapsed_us` will have the same type in this expression, but `data->timeout_ms * 1000` converted to `uint32_t`

Reply:
Yes, and then it will be stored into `int32_t` variable. You will get large negative value there in some cases